azurelinuxagent/ga/exthandlers.py

```diff
@@ -959,52 +957,32 @@ def report_ext_handlers_status(self, incarnation_changed=False, vm_agent_update_
                       message=msg)
             return None
 
-    def write_ext_handlers_status_to_info_file(self, vm_status, incarnation_changed):
-        status_file = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
-
-        if os.path.exists(status_file) and incarnation_changed:
-            # On new goal state, move the last status report for the previous goal state to the history folder
-            last_modified = os.path.getmtime(status_file)
-            timestamp = datetime.datetime.utcfromtimestamp(last_modified).isoformat()
-            GoalStateHistory(timestamp, "status").save_status_file(status_file)
-
-        # Now create/overwrite the status file; this file is kept for debugging purposes only
+    def get_ext_handlers_status_debug_info(self, vm_status):
         status_blob_text = self.protocol.get_status_blob_data()
         if status_blob_text is None:
             status_blob_text = ""
 
-        debug_info = ExtHandlersHandler._get_status_debug_info(vm_status)
-
-        status_file_text = \
-'''{{
-    "__comment__": "The __status__ property is the actual status reported to CRP",
-    "__status__": {0},
-    "__debug__": {1}
-}}
-'''.format(status_blob_text, debug_info)
-
-        fileutil.write_file(status_file, status_file_text)
```

Comment:
now the file is created by the caller, and this code is exposed as get_ext_handlers_status_debug_info